### PR TITLE
[PT] StreamData translation

### PR DESCRIPTION
**Pull request by batistaleandro**
[PT] New translation: StreamData #2392

### CODE REVIEW

File: pt/lessons/libraries/stream-data.md [hunk 1]
```diff
@@ -0,0 +1,457 @@
+---
+version: 1.1.0
+title: StreamData
+---
+
+Uma biblioteca  de testes unitários baseada em exemplos como  [ExUnit](https://hexdocs.pm/ex_unit/ExUnit.html) é uma excelente ferramente para te ajudar a verifica se seu código funciona da forma como você imagina que ele funciona. Contudo, testes unitários baseados em exemplos têm algumas desvantagens:
```
Comment (vbrazo) [inline pt/lessons/libraries/stream-data.md:6]:
```suggestion
Uma biblioteca de testes unitários baseada em exemplos como  [ExUnit](https://hexdocs.pm/ex_unit/ExUnit.html) é uma excelente ferramente para te ajudar a verifica se seu código funciona da forma como você imagina que ele funciona. Contudo, testes unitários baseados em exemplos têm algumas desvantagens:
```

File: pt/lessons/libraries/stream-data.md [hunk 2]
```diff
@@ -0,0 +1,457 @@
...
+
+- Pode ser fácil esquecer alguns casos extremos, uma vez que você só está testando alguns inputs.
+- Você pode escreve testes sem pensar nos requerimentos de forma completa.
+- Estes testes podem ser verbosos quando você utiliza vários exemplos para uma função.
+
+Nesta aula iremos apreder como [StreamData](https://github.com/whatyouhide/stream_data) can pode nos ajudar a superar esses pontos negativos.
```
Comment (vbrazo) [inline pt/lessons/libraries/stream-data.md:12]:
```suggestion
Nesta aula iremos aprender como [StreamData](https://github.com/whatyouhide/stream_data) pode nos ajudar a superar esses pontos negativos.
```

File: pt/lessons/libraries/stream-data.md [hunk 3]
```diff
@@ -0,0 +1,457 @@
...
+
+{% include toc.html %}
+
+## O que é StreamData?
+
+[StreamData](https://github.com/whatyouhide/stream_data) é uma biblioteca que executa testes sem estado baseados em propriedade .
+
+A bilbioteca StreamData irá executar cada teste [100 vezes por padrão](https://hexdocs.pm/stream_data/ExUnitProperties.html#check/1-options), usando dados aleatórias a cada vez. Quando um teste falhar, StreamData irá tentar [encolher](https://hexdocs.pm/stream_data/StreamData.html#module-shrinking) a entrada para o menor valor capaz de fazer com que o teste falhe. Isso pose ser útil quando você tem que debugar seu código! Se uma lista de 50 elementos faz com que sua função falhe, e somente um dos elementos da lista é problemático, StreamData pode te ajudar a identificar o elemento detrator.
```
Comment (vbrazo) [inline pt/lessons/libraries/stream-data.md:20]:
```suggestion
A biblioteca StreamData irá executar cada teste [100 vezes por padrão](https://hexdocs.pm/stream_data/ExUnitProperties.html#check/1-options), usando dados aleatórias a cada vez. Quando um teste falhar, StreamData irá tentar [encolher](https://hexdocs.pm/stream_data/StreamData.html#module-shrinking) a entrada para o menor valor capaz de fazer com que o teste falhe. Isso pose ser útil quando você tem que debugar seu código! Se uma lista de 50 elementos faz com que sua função falhe, e somente um dos elementos da lista é problemático, StreamData pode te ajudar a identificar o elemento detrator.
```

File: pt/lessons/libraries/stream-data.md [hunk 4]
```diff
@@ -0,0 +1,457 @@
+---
+version: 1.1.0
+title: StreamData
+---
+
+Uma biblioteca  de testes unitários baseada em exemplos como  [ExUnit](https://hexdocs.pm/ex_unit/ExUnit.html) é uma excelente ferramente para te ajudar a verifica se seu código funciona da forma como você imagina que ele funciona. Contudo, testes unitários baseados em exemplos têm algumas desvantagens:
+
+- Pode ser fácil esquecer alguns casos extremos, uma vez que você só está testando alguns inputs.
+- Você pode escreve testes sem pensar nos requerimentos de forma completa.
```
Comment (vbrazo) [inline pt/lessons/libraries/stream-data.md:9]:
```suggestion
- Você pode escrever testes sem pensar nos requerimentos de forma completa.
```

File: pt/lessons/libraries/stream-data.md (hunk 3 above)
Comment (vbrazo) [inline pt/lessons/libraries/stream-data.md:20]:
```suggestion
A bilbioteca StreamData irá executar cada teste [100 vezes por padrão](https://hexdocs.pm/stream_data/ExUnitProperties.html#check/1-options), usando dados aleatórios a cada vez. Quando um teste falhar, StreamData irá tentar [encolher](https://hexdocs.pm/stream_data/StreamData.html#module-shrinking) a entrada para o menor valor capaz de fazer com que o teste falhe. Isso pode ser útil quando você tem que debugar seu código! Se uma lista de 50 elementos faz com que sua função falhe, e somente um dos elementos da lista é problemático, StreamData pode te ajudar a identificar o elemento detrator.
```

**Comment by vbrazo** [comment]
@batistaleandro quick review. Thanks for doing this 🚀

**Comment by samantadearaujo** [comment]
Nice!

**Review by samantadearaujo — DISMISSED** [review]

**Comment by doomspork** [comment]
@batistaleandro @samantadearaujo @vbrazo can this be converted to a PR and merged?

**Comment by anderkonzen** [comment]
Hi @batistaleandro ! This PR is a bit old. Do you mind going through the original EN lesson to check if newer changes were introduced?

**Comment by doomspork** [comment]
@batistaleandro are you interested in updating this lesson or should we close the PR? We've moved to a new lesson format for the Phoenix site so we'll need to make those changes to this PR too.

**Comment by anderkonzen** [comment]
Hi @doomspork ! I re-checked the PR and more than half of the lesson is still not translated. I suggest we close this PR for now, and @batistaleandro I think you can re-open it after the migration to the new Phoenix site.

**Pull request closed without merging**